modules/tinymce/src/core/main/ts/fmt/ExpandRange.ts

```diff
@@ -121,14 +121,14 @@ const findSelectorEndPoint = (dom: DOMUtils, formatList: Format[], rng: Range, c
 };
 
 const findBlockEndPoint = (editor: Editor, formatList: Format[], container: Node, siblingName: Sibling) => {
-  let node = container;
+  let node: Node | null = container;
   const dom = editor.dom;
   const root = dom.getRoot();
   const format = formatList[0];
 
   // Expand to block of similar type
-  if (FormatUtils.isBlockFormat(format) && !format.wrapper) {
-    node = dom.getParent(container, format.block, root);
+  if (FormatUtils.isBlockFormat(format)) {
+    node = format.wrapper ? null : dom.getParent(container, format.block, root);
```

Comment:
Turns out this needs to be falsy when using a block wrapper so that it flows into the logic on line 134. Without that it fails to expand to the block and only ends up doing a word grab for the current word.